### ISSUE TITLE
Indent forces other arguments to start with no indent if the first argument is multiline

### ISSUE THREAD

**Issue by Alexsey** [eslint/eslint#10389]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 8.10
* **npm Version:** 5.6

**What parser (default, Babel-ESLint, etc.) are you using?** -

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
"indent": [1, 4, {"SwitchCase": 1}]
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function foo () {}

// error
foo([
        1
    ],
    true,
    false
)
// no error
foo([
    1
],
true,
false)
```

**What did you expect to happen?**
There should be no error in the first case. Even more - the second case should be an error because it's not ok to have single line arguments (`true` and `false`) on the same indentation level as function invocation. I understand that
```js
f([
    1
])
```
is a valid case and
```js
f([
    1
], {
    x: 2
})
```
may also be considered as a valid one but only as long as there are no single line arguments! Or even streaker - only closes of arguments wrappers (like `}` and `]`) may have the same indentation as an invocation and if any argument is not satisfying this rule then they all should be indented. This would disallow
```js
foo([
    1
2], {
    x: 1,
y: 2})
```
but would allow
```js
foo([1,
    2
], {x: 1,
    y: 2
})
```

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint: Expected indentation of 4 but found 8. (indent)

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report.

However, this is working as intended. The intent is that code like this is considered correct:

```js
foo({
    a: 1
},
{
    a: 2
});
```

Unfortunately, I think this is a case where different people have a lot of different preferences on how something should be indented, and the `indent` rule happens to enforce this particular preference.

**Comment by Alexsey** [comment]
@not-an-aardvark I have seen that your example is considered to be correct and I'm not arguing against of it. I'm sorry I have given the wrong explanation of the edge cases to correct examples - if multiline argument is started from `[` or `{` it is always ok. Not ok if multiline argument is ended not with `{` or `[`.

The key points are:
* there should be a way to have an intended argument list that is started from multiline argument - now it considered to be invalid 
* it should not be ok have a mix of unintented multiline and singleline arguments

**Comment by kaicataldo** [comment]
@Alexsey The indent rule is _very_ complicated and we have to be judicious about any changes we make, since the maintenance burden can be quite high. That being said, we're always willing and evaluate concrete proposals for changes/improvements. If you'd like to create such a proposal, please follow the instructions here: https://eslint.org/docs/developer-guide/contributing/rule-changes. Thanks!

**Comment by Alexsey** [comment]
@kaicataldo should I split this issue into two with one about the allowance of indented multiline arguments list where the first argument is multiline and second about forbidden of non-indented multiline arguments list with a mix of single line and multiline arguments?

**Comment by kaicataldo** [comment]
If you think these are two separate issues, it’s always nicer to have smaller proposals and PRs, as they’re much more likely to be resolved quickly and are easy to understand.

**Comment by Alexsey** [comment]
Closed in favor of https://github.com/eslint/eslint/issues/10417 and https://github.com/eslint/eslint/issues/10418
